### ISSUE TITLE
Merge PRs that contains certain label

### ISSUE THREAD

**Issue by rafaelstz** [gr2m/merge-schedule-action#91]
Sometimes you just want to merge by scheduling the PRs already approved. It can be done using Github labels.

**Comment by cameronleger** [comment]
Hi @gr2m, I am willing to contribute this feature, if you agree with it. This Action seems like the best fit for my uses except for this one feature request.

I'm sure I can meet the requirements of the task. I'm only reaching out to see if you have any requirements or recommendations for the design. Some should be obvious, like good tests; I'm just trying to avoid wasting your time with rounds of feedback or PRs/Issues being open too long.

For example, allowing the requirement of _multiple_ labels is not in my needs, but it is certainly more flexible and future-safe; given the available input types, there's a deserialization decision to be made: how would you prefer to accept those labels in a string, if you would support multiple labels?

I don't need answers - I'm good to simply do the work I believe in and open a PR for that. The implementation seems quite easy.

**Comment by gr2m** [comment]
I don't quite understand what this feature will do? Do you want to add a `required-label` input or similar and it will only merge if said label is set on the pull request?

**Comment by cameronleger** [comment]
Exactly right. I can provide a use case if that helps make it clearer.

I have a few schedules for generating PRs, as well as ad-hoc PRs falling into the same few "labels." Although their contents are in the same repository, what these labels represent are an important dimension for most purposes of the repository: the files, processes, and PRs are sometimes quite related to their label. It would be much harder to turns these "labels" into separate repositories, because most usage patterns are aligned with this single repository instead of the label.

I'm looking at the opposite end of the process now: the PRs deserve schedules for merging just as well as creation. _When_ these schedules occur are significantly more related to these labels than to their creation. As such, it's duplicative and potentially wrong, even though possible, for the creators of the PRs to determine when they should be merged. It's more accurately stated that "label X" must be merged every Monday, "label Y" every other Wednesday, and so on. Given the multiple producers, machine and human alike, setting schedules at this consumer end is better for these purposes.

In other words, sometimes there are multiple schedules for the same repository, but each one applies to a subset of the open/valid PRs. Using labels to categorize these seems to be the most reasonable approach given the constraints. Alternatively, I know that restructuring the repository such that the schedules align with the contents exactly is possible; I'm just not willing to forgo the benefits of their current structures.

**Comment by gr2m** [comment]
so you would only use the label as indicator whether to merge a PR, and wouldn't use the `/schedule` command at all? I feel like that would go outside the scope of this action. But maybe send your PR, worst case you can simply point to your fork at `cameronleger/merge-schedule-action` instead of `gr2m/merge-schedule-action` to get the functionality you want

**Comment by cameronleger** [comment]
I thought it would be conjunctive, so `/schedule` is still totally necessary and functions as it does today. With the `required_labels: "bug"` input specified, then the scheduling action here would only merge what is already considered mergeable _and_ has the "bug" label.

If it only considered the labels, then I agree that this is not the action for that. At first glance, the `/schedule` command _seems_ to be exactly when the PR _will be_ merged; of course, given the capabilities of Actions, it behaves more like when the PR _can be_ merged on/after. It's truly the cron-job that determines _when_ the merge happens, and the command only _filters_ what can be merged at that time. So, this additional filter does not seem to compete with that command or the goals of this Action. I wouldn't push hard for it, but I think that the required text in the PR's description is more accurately stated like `/enable-auto-merge` and `/enable-auto-merge-after 2024-08-28`.

I could see confusion arising from mis-configurations, as they do today, so it would need good documentation. Perhaps this will explain the intent and behavior better:
```yaml
jobs:
  merge_schedule:
    runs-on: ubuntu-latest
    steps:
      - uses: gr2m/merge-schedule-action@v2
        with:
          # Merge method to use. Possible values are merge, squash or
          # rebase. Default is merge.
          merge_method: squash
          # Time zone to use. Default is UTC.
          time_zone: 'America/Los_Angeles'
          # Require all pull request statuses to be successful before
          # merging. Default is `false`.
          require_statuses_success: 'true'
          # Consider only pull requests matching all of these labels.
          # Values are comma-separated; empty string disables check.
          # Default is '`, i.e. disabled.
          require_labels: 'bug,minor'
          # Label to apply to the pull request if the merge fails. Default is
          # `automerge-fail`.
          automerge_fail_label: 'merge-schedule-failed'
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by gr2m** [comment]
thank you for explaining it to me. I'm okay with adding the `require_labels` input, I think users who want to use it understand the changed meaning of the `/schedule` command. And it won't cause friction to existing users or users who don't need the feature. So yes, please send your PR 👍
